olive/common/ort_inference.py

```diff
@@ -37,6 +38,43 @@ def ort_supports_ep_devices() -> bool:
     return version.parse(OrtVersion).release >= version.parse("1.23.0").release
 
 
+def maybe_add_winml_ep_paths(ep_paths: dict[str, str]) -> dict[str, str]:
+    """Get the Windows ML EP paths."""
+    try:
+        from onnxruntime import winml as _  # noqa: F401 # pylint: disable=unused-import
```

Comment:
this will not work. onnxruntime-winml GA is same as onnxruntime-dml.

We should check if we are in Windows and we support `register_execution_provider_library`, then try this. And if we met error, we may be advice user to use onnxruntime-winml in Windows with proper app sdk version and Windows version etc.